lib/core/filtering/post_process_result.rb

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+module Core
+  module Filtering
+    class PostProcessResult
+      attr_reader :document, :matching_rule
+
+      def initialize(document, matching_rule)
+        @document = document
+        @matching_rule = matching_rule
+      end
+
+      def policy
+        matching_rule.policy
+      end
+
+      def include?
```

Comment:
Oops I [named these a bit differently](https://github.com/elastic/connectors-ruby/blob/main/lib/connectors/base/simple_rules_parser.rb#L39-L45). =) Should we sync up?

Reply:
nah, I'll just adopt your approach. Since your changes merged already, I think it will be easy for me to rebase on main and then refactor what I have here to unify with your changes.